### ISSUE TITLE
[build] Feature Switch to Enable PIC Support

### ISSUE THREAD

**Issue by ppenna** [ppenna/nanvix#383]
## Description

Currently we employ a best-effort strategy to enable the PIC at startup time:
- If LAPIC is supported, we try to initialize it.
- If we fail, or if LAPIC is not supported, we try to enable the 8259 PIC.
- If we fail, we turn off PIC support.

We should have a feature switch that disables PIC support at build time, to avoid higher startup times on platforms that don't support interrupts at all.

## Code Snippets

The code snippet bellows points to where we first detect that interrupts are not supported.

https://github.com/nanvix/kernel/blob/cc7d169cacb92f5cf97c7e1700222de7579ab930/src/hal/mod.rs#L77-L85

**Comment by lemosep** [comment]
The feature must be pushed added into the cc7d169 branch?

**Comment by ppenna** [comment]
> The feature must be pushed added into the [cc7d169](https://github.com/nanvix/kernel/commit/cc7d169cacb92f5cf97c7e1700222de7579ab930) branch?

No. Every PR should be opened towards latest `dev` branch. Permalink was used to link the concerned code-snippet here.

**Comment by ppenna** [comment]
@lemosep are you working on this issue?

**Comment by ppenna** [comment]
Removed previous asignees due to inactivity.
